api/v1beta1/common.go

```diff
@@ -102,3 +105,15 @@ type NetworksSection struct {
 	// FixedIP - Specific IP address to use for this network
 	FixedIP string `json:"fixedIP,omitempty" yaml:"fixedIP,omitempty"`
 }
+
+// AssertNodeAndRoleUniqueness - assert uniqueness between node and role fields
```

Comment:
`AssertNodeandRoleUniqueness` and `AssertNodeAndNodeUniqueness` could both work on taking the different types as input (role or node) and extracting the items to compare (`RunnerImage` or `NetworkAttachments`) but then both could pass to the same third function (e.g. `AssertUniqueness`) which ensures that the `RunnerImage` or `NetworkAttachments` do not differ.

If the goal is to not have duplicate code that's one way I see to do it.

Reply:
I'll try